python/GafferSceneUI/SetUI.py

```diff
@@ -193,7 +193,7 @@ def __setsPopupMenu( menuDefinition, plugValueWidget ) :
 			insertAt = (cursorPosition, cursorPosition)
 
 	node = plug.node()
-	if isinstance( node, GafferScene.Filter ) :
+	if isinstance( node, (GafferScene.Filter, GafferScene.ObjectSource) ) :
```

Comment:
What's the intention here? The general meaning of the "ui:scene:acceptsSetExpression" metadata is that a menu should be shown for sets available as inputs to the node (or filter). In this case, there are no such sets, and all we're achieving is to list sets that are output from the ObjectFilter, none of which are useful for the filteredLights attribute as far as I can see. There's also an open ticket at IE that is going to require some changes to `__downstreamNodes()` so that it becomes much more filter-specific. Unless I'm missing something, we should drop this and also remove the "ui:scene:acceptsSetExpression" metadata from the LightFilter node.

Reply:
My intention was to provide the menu in a graph like this. We don't have to, but I thought it'd be a nice touch and easy to do. I wasn't aware of the other ticket.

```
import Gaffer
import GafferArnold
import GafferScene
import IECore
import imath

__children = {}

__children["light_blocker"] = GafferArnold.ArnoldLightFilter( "light_blocker" )
parent.addChild( __children["light_blocker"] )
__children["light_blocker"].loadShader( "light_blocker" )
__children["light_blocker"].addChild( Gaffer.V2fPlug( "__uiPosition", defaultValue = imath.V2f( 0, 0 ), flags = Gaffer.Plug.Flags.Default | Gaffer.Plug.Flags.Dynamic, ) )
__children["Parent"] = GafferScene.Parent( "Parent" )
parent.addChild( __children["Parent"] )
__children["Parent"].addChild( Gaffer.V2fPlug( "__uiPosition", defaultValue = imath.V2f( 0, 0 ), flags = Gaffer.Plug.Flags.Default | Gaffer.Plug.Flags.Dynamic, ) )
__children["spot_light"] = GafferArnold.ArnoldLight( "spot_light" )
parent.addChild( __children["spot_light"] )
__children["spot_light"].loadShader( "spot_light" )
__children["spot_light"].addChild( Gaffer.V2fPlug( "__uiPosition", defaultValue = imath.V2f( 0, 0 ), flags = Gaffer.Plug.Flags.Default | Gaffer.Plug.Flags.Dynamic, ) )
__children["Set"] = GafferScene.Set( "Set" )
parent.addChild( __children["Set"] )
__children["Set"].addChild( Gaffer.V2fPlug( "__uiPosition", defaultValue = imath.V2f( 0, 0 ), flags = Gaffer.Plug.Flags.Default | Gaffer.Plug.Flags.Dynamic, ) )
__children["light_blocker"]["parameters"]["density"].setValue( 1.0 )
__children["light_blocker"]["__uiPosition"].setValue( imath.V2f( 0.94999975, -1.66406333 ) )
__children["Parent"]["in"].setInput( __children["light_blocker"]["out"] )
__children["Parent"]["parent"].setValue( '/' )
__children["Parent"]["child"].setInput( __children["Set"]["out"] )
__children["Parent"]["__uiPosition"].setValue( imath.V2f( 12.9491501, -9.82812595 ) )
__children["spot_light"]["__uiPosition"].setValue( imath.V2f( 22.0491486, 6.5 ) )
__children["Set"]["in"].setInput( __children["spot_light"]["out"] )
__children["Set"]["name"].setValue( 'bestLights' )
__children["Set"]["__uiPosition"].setValue( imath.V2f( 22.0491486, -1.6640625 ) )

del __children

```

Reply:
Hmm, it is handy in that example, but it's really fragile - stick a Group under the light_blocker and before the Parent and it doesn't work at all. I do think we're better off backing out of this one I'm afraid...

Reply:
Done in 3125b5c. Let me know if you want me to drop that commit entirely.